src/main/java/Duke.java

```diff
@@ -1,10 +1,147 @@
+import java.util.Scanner;
+import java.util.Arrays;
+
 public class Duke {
+
+    public static final int MAX_TASK = 100;
+    public static final String COMMAND_LIST = "list";
+    public static final String COMMAND_MARK = "mark";
+    public static final String COMMAND_UNMARK = "unmark";
+    public static final String COMMAND_BYE = "bye";
+    public static final String COMMAND_EVENT = "event";
+    public static final String COMMAND_DEADLINE = "deadline";
+    public static final String COMMAND_TODO = "todo";
+
+    private static Task[] taskItems;
+    private static int itemNum;
+    private static final Scanner SCANNER = new Scanner(System.in);
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+
+        welcomeMessage();
+        initTaskTable();
+
+        while (true) {
+            String inCommand = SCANNER.nextLine();
+            executeCommand(inCommand);
+        }
+
+    }
+
+    private static void executeCommand(String inCommand) {
+        String[] commandArr = inCommand.split(" ");
+        int listNum;
+        switch (commandArr[0]) {
+        case COMMAND_LIST:
+            listNum = 1;
+            if (itemNum == 0) {
+                System.out.println("List is empty");
+            }
+            else {
+                for (int i = 0; i < itemNum; i++) {
+                    System.out.println(listNum + ".[" + taskItems[i].getTaskType() + "]" + "[" + taskItems[i].getStatusIcon() + "] " + taskItems[i].description);
+                    listNum++;
+                }
+            }
+            break;
+        case COMMAND_MARK:
+            int markNum = Integer.parseInt(commandArr[1]) - 1;
+            taskItems[markNum].isDone = true;
+            System.out.println("Nice! I've marked this task as done:\n" + "[" + taskItems[markNum].taskType + "]"+ "[X] " + taskItems[markNum].description);
+            break;
+        case COMMAND_UNMARK:
+            int unMarkNum = Integer.parseInt(commandArr[1]) - 1;
+            taskItems[unMarkNum].isDone = false;
+            System.out.println("Nice! I've marked this task as not done yet:\n" + "[" + taskItems[unMarkNum].taskType + "]" + "[ ] " + taskItems[unMarkNum].description);
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            addTask(inCommand);
+            break;
+        }
+    }
+
+    private static void exitProgram() {
+        System.out.println("Bye. Hope to see you again soon!");
+        System.exit(0);
+    }
+
+    private static void addTask(String inCommand) {
+        String[] sentenceArr = inCommand.split(" ");
+        String[] taskToInput;
```

Comment:
Would be better to have a plural array name since it is a string array, maybe you can change to tasksToInputs